services/checks/funcs.go

```diff
@@ -80,3 +82,48 @@ func formatVersionNum(args ...interface{}) (interface{}, error) {
 	}
 	return p.String(), nil
 }
+
+func GetAdditionalContext() map[string]starlark.GoFunc {
+	return map[string]starlark.GoFunc{
+		"ip_is_private": ipIsPrivate,
+	}
+}
+
+var privateAddressBlocks = []string{
+	// private blocks, see https://tools.ietf.org/html/rfc1918
+	"10.0.0.0/8",
+	"172.16.0.0/12",
+	"192.168.0.0/16",
+	// link-local block, see https://tools.ietf.org/html/rfc3927
+	"169.254.0.0/16",
+	// loop-back block, see https://tools.ietf.org/html/rfc5735
+	"127.0.0.0/8",
+}
+
+// ipIsPrivate accepts a single string argument (IP address) and
+// returns true for a private address, otherwise false.
+func ipIsPrivate(args ...interface{}) (interface{}, error) {
+	if l := len(args); l != 1 {
+		return nil, errors.Errorf("expected 1 argument, got %d", l)
+	}
+
+	ip, ok := args[0].(string)
+	if !ok {
+		return nil, errors.Errorf("expected string argument, got %[1]T (%[1]v)", args[0])
+	}
+
+	ipAddress := net.ParseIP(ip)
```

Comment:
If we pass not an ip address, for example string "test" as argument, that function returns `false`. But in fact passed argument if wrong at all. Should we check it?

Reply:
we could add a check, if `ParseIP` returns  nil in such cases we return an error saying "invalid ip"

Reply:
Yep, I think to have an error is more clear in that case

Reply:
⚠️ The only problem is that this error can't be handled in the script itself, and that might be useful. So let's log error using environment's debug logger and return `nil`/`None` in that case. That will require to change that function to a method.

I updated ticket description to mention that,

Reply:
i can log the error instead of returning it, but i can't understand this part

> That will require to change that function to a method.

which function are you talking about `ipIsPrivate` and change it to a method over what type?

Reply:
What logger you are going to use?

Reply:
i thought you meant to log it with just a debug log level but I think i misunderstood that you mean "some logger that we already use somewhere"  by this `environment's debug logger` to be used here??? I am not sure :confused:

Reply:
Doh, I renamed it to `print` and forgot about it :)

https://github.com/percona-platform/platform/blob/27c08746e97b7af95712d5361053b968718fc4d2/pkg/starlark/starlark.go#L130-L133

We should use that PrintFunc. But accessing it would require something like:
```go
func (env *Env) ipIsPrivate …
```

But we can't do that there. So yeah, let's use logrus with Error level

Reply:
done.